micromanager_gui/main_window.py

```diff
@@ -594,8 +603,15 @@ def update_max_min(self, event=None):
 
     def snap(self):
         self.stop_live()
-        self._mmc.snapImage()
-        self.update_viewer(self._mmc.getImage())
+
+        # snap in a thread so we don't freeze UI when using process local mmc
+        @thread_worker(
+            connect={"finished": lambda: self.update_viewer(self._mmc.getImage())}
+        )
+        def snap_worker():
+            self._mmc.snapImage()
+
+        snap_worker()
```

Comment:
```suggestion
        # snap in a thread so we don't freeze UI when using process local mmc
        create_worker(
            self._mmc.snapImage,
            _connect={"finished": lambda: self.update_viewer(self._mmc.getImage())},
            _start_thread=True,
        )
```

can use a lower-level function here (lower than the thread_worker decorator)  use `from superqt.utils import create_worker`